k-distribution/k-tutorial/1_basic/07_side_conditions/README.md

```diff
@@ -1 +1,171 @@
+    requires I >Int 0
+endmodule
+```
+
+In this case, the `isPositive` function takes a single integer argument. The
+function evaluates to true if the argument passed is greater than zero. Note
+that the side condition is allowed to refer to variables that appear on the
+left-hand-side of the rule. In the same manner as variables appearing on the
+right-hand-side, variables that appear in the side condition evaluate to the
+value that was matched on the left hand side. Then the functions in the
+side condition are evaluated, which returns a term of sort `Bool`. If the term
+is equal to `true`, then the rule applies. Bear in mind that the side condition
+is only evaluated at all if the patterns on the left-hand-side of the rule
+match the term being evaluated.
+
+### Exercise
+
+Write a rule that evaluates `isPositive` to `false` if the argument to the
+function is zero or negative. Test that the function correctly evaluates 
+various positive, zero, and negative integers.
+
+## `owise` Rules
+
+So far, all the rules we have introduced have had the same **priority**. What
+this means is that K does not necessarily enforce an order in which the rules
+are tried. We have only discussed functions so far in K, so it is not
+immediately clear why this choice was made, given that a function is not
+considered well-defined if multiple rules for evaluating it are capable of
+evaluating the same arguments to different results. However, in future lessons
+we will discuss other types of rules in K, some of which can be
+**nondeterministic**. What this means is that if more than one rule is capable
+of matching, then K will explore both possible rules in parallel, and consider
+each of their respective results when executing your program. Don't worry too
+much about this right now, but just understand that because of the potential
+later for nondeterminism, we don't enforce a total ordering on the order in
+which rules are attempted to be applied.
+
+However, sometimes this is not practical; It can be very convenient to express
+that a particular rule applies if no other rules for that function are
+applicable. This can be expressed by adding the `owise` attribute to a rule.
+What this means, in practice, is that this rule has lower priority than other
+rules, and will only be tried to be applied after all the other,
+higher-priority rules have been tried and they have failed.
+
+For example, in the above exercise, we had to add a side condition to the rule
+we wrote to handle `isPositive` for negative numbers. However, in practice this
+meant that the integer comparison operation happened twice. We can more
+efficiently and more idiomatically write the negative case for the `isPositive`
+rule using the `owise` attribute (`lesson-07-b.k`):
+
+```k
+module LESSON-07-B
+  imports BOOL
+  imports INT
+
+  syntax Bool ::= isPositive(Int) [function]
+
+  rule isPositive(I) => true
+    requires I >Int 0	
+  rule isPositive(_) => false [owise] 
```

Comment:
```suggestion
  rule isPositive(I) => true  requires I >Int 0	
  rule isPositive(_) => false [owise] 
```

I like to write function definitions with `[owise]` so that it lines up with the requires clauses, somehow it reads better in my mind.

Reply:
Your choice though.